src/Compilers/Core/Portable/Operations/Loops/IDoLoopStatement.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a C# 'do while' or VB 'Do While' or 'Do Until' loop statement.
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IDoLoopStatement : ILoopStatement
+    {
+        /// <summary>
+        /// Condition of the loop.
+        /// </summary>
+        IOperation Condition { get; }
+
+        /// <summary>
+        /// Represents kind of do loop statement.
+        /// </summary>
+        DoLoopKind DoLoopKind { get; }
+
+        /// <summary>
+        /// Additional conditional supplied for loop in error cases.
+        /// For example, for VB 'Do While' or 'Do Until' loop with syntax errors where both the top and bottom conditions are provided.
+        /// The top condition is preferred and exposed as <see cref="Condition"/> and the bottom condition is exposed by this property.
+        /// This property should be null for all non-error cases.
+        /// </summary>
+        IOperation InvalidCondition { get; }
```

Comment:
>InvalidCondition [](start = 19, length = 16)

I think we should come up with a better name. The condition on its own could be a valid expression.

Reply:
I am thinking `IgnoredCondition` or `AdditionalCondition`. Any suggestions or preferences @dotnet/analyzer-ioperation ?

Reply:
I am going to change this to `IgnoredCondition`, let me know if anyone has any objections.